common/build/eslint-plugin-fluid/src/test/mockFiles/no-unchecked-record-access/fileWithOnlyRecordAccess.ts

```diff
@@ -16,57 +16,61 @@ type StaticType = { a: string; b: string };
 someObjWithStaticType[b].length; // ok: Accessing length of string property 'b' using bracket notation
-someObjWithStaticType[c].length; // defect: Accessing length of undefined property 'c' using bracket notation
-const aExpectingStringFromStaticType: string = someObjWithStaticType.a; // ok: Assigning string property to a strict string variable, 'a' is defined
+someObjWithStaticType[c].length; // defect: Accessing length of missing property 'c' using bracket notation
+const aExpectingStringFromStaticType: string = someObjWithStaticType.a; // ok: Assigning string property to a strict string variable
 const aExpectingStringOrUndefinedFromStaticType: string | undefined = someObjWithStaticType.a; // ok: Assigning string property to a string or undefined variable
 
-const someObjWithPotenciallyUndefinedProperties: IndexSignatureType = someObjWithStaticType;
-someObjWithPotenciallyUndefinedProperties.a; // ok: Accessing dynamic property 'a', its type is dynamic
-someObjWithPotenciallyUndefinedProperties.a.length; // defect: Accessing length of dynamic property 'a', TypeScript can't guarantee 'a' is a string
-someObjWithPotenciallyUndefinedProperties["a"]; // ok: Accessing property 'a' using bracket notation
-someObjWithPotenciallyUndefinedProperties["a"].length; // defect: Accessing length of potencially undefined property 'a' using bracket notation
-someObjWithPotenciallyUndefinedProperties[a].length; // ok: Accessing length of string property 'a' using bracket notation
-someObjWithPotenciallyUndefinedProperties[b].length; // ok: Accessing length of string property 'b' using bracket notation
-someObjWithPotenciallyUndefinedProperties[c].length; // defect: Accessing length of undefined property 'c' using bracket notation
-someObjWithPotenciallyUndefinedProperties.a?.length; // ok: Using optional chaining to access length, safely handles 'undefined'
+const indexedRecordOfStrings: IndexSignatureType = someObjWithStaticType;
+indexedRecordOfStrings.a; // ok: Accessing index property 'a' without requiring a particular result
+indexedRecordOfStrings.a.length; // defect: Accessing length of index property 'a', but 'a' might not be present
+indexedRecordOfStrings["a"]; // ok: Accessing property 'a' using bracket notation
+indexedRecordOfStrings["a"].length; // defect: Accessing length of index property 'a' using bracket notation, but 'a' might not be present
+indexedRecordOfStrings[a].length; // defect: Accessing length of index property 'a' using bracket notation, but 'a' might not be present
+indexedRecordOfStrings[b].length; // defect: Accessing length of index property 'b' using bracket notation, but 'b' might not be present
+indexedRecordOfStrings[c].length; // defect: Accessing length of missing property 'c' using bracket notation
+indexedRecordOfStrings.a?.length; // ok: Using optional chaining to access length safely handles 'undefined'
+indexedRecordOfStrings.a!.length; // ok: Using non null assert to access length safely handles 'undefined'
```

Comment:
This does not safely handle `undefined`. The author says they understand the question raised by check and acknowledge that they have other information expecting that it is actually defined or that they are okay with an exception being raise here if "a" is not present and defined.